ObjectCreation/AeroFoil.FCMacro

```diff
@@ -0,0 +1,1747 @@
+                            self.profileName
+                        ).addConstraint(
+                            Sketcher.Constraint(
+                                "DistanceY", -1, 1, j, 3, self.points[1][j]
+                            )
+                        )
+                        App.ActiveDocument.getObject(self.profileName).setDatum(
+                            constraintIndex,
+                            App.Units.Quantity(str(self.points[1][j]) + "mm"),
+                        )
+                        if self.designprogressbar.value() < 75:
+                            self.designprogressbar.setValue(
+                                round(70 + ((j / (20 * n_)) * 100))
+                            )
+            elif self.designtype[0] == 1:
+                DWireOrBSpline_, pointsVector = 0, []
+                n_ = len(self.points[0])
+                for i in range(n_):
+                    pointsVector.append(
+                        FreeCAD.Vector(self.points[0][i], 0, self.points[1][i])
+                    )
+                    if self.designprogressbar.value() < 75:
+                        self.designprogressbar.setValue(
+                            round(50 + ((i / (4 * n_)) * 100))
+                        )
+                if self.designtype[1] == 1:
+                    # if self.designclosed is determined by face=self.designclosed
+                    DWireOrBSpline_ = Draft.makeWire(
+                        pointsVector,
+                        closed=True,
+                        placement=None,
+                        face=self.designclosed,
+                        support=None,
+                    )
+                elif self.designtype[1] == 2:
+                    # if self.designclosed is determined by face=self.designclosed
+                    DWireOrBSpline_ = Draft.makeBSpline(
+                        pointsVector,
+                        closed=True,
+                        placement=None,
+                        face=self.designclosed,
+                        support=None,
+                    )
+                App.activeDocument().recompute()
+                shape_ = App.ActiveDocument.addObject("Part::Feature", self.profileName)
+                shape_.Shape = DWireOrBSpline_.Shape
+                App.ActiveDocument.removeObject(DWireOrBSpline_.Label)
+        except Exception:
+            # Unexpected Error Occurred while Processing
+            try:
+                App.ActiveDocument.removeObject(self.profileName)
+            except Exception:
+                pass
+            App.activeDocument().recompute()
+            return False
+        App.activeDocument().recompute()
+        FreeCADGui.activeDocument().activeView().viewFront()
+        FreeCADGui.SendMsgToActiveView("ViewFit")
+        return True
+
+
+
+# Sub-main Function Class
+# ------------------------------------------------------------------------------------------------
+
+class AeroFoilDialog:
```

Comment:
This could be in a separate file.

Reply:
I'll leave that for the next version. I've pondered upon your idea of adding a python command as well. Hence, it would make sense to re-arrange the code then. It's bit of a hassle at the moment.